Source/ACE/Factories/GenericObjectFactory.cs

```diff
@@ -26,7 +27,9 @@ public static List<WorldObject> CreateWorldObjects(List<AceObject> sourceObjects
                 }
                 else if ((oDescFlag & ObjectDescriptionFlag.Portal) != 0)
                 {
-                    results.Add(new Portal(aceO));
+                    AcePortalObject acePO = DatabaseManager.World.GetPortalObjectsByAceObjectId(aceO.AceObjectId);
```

Comment:
This brings up a design flaw I have that I need to address.  Ugh.  This will certainly work, though.